devops/providers/azure-devops/templates/infrastructure/README.md

```diff
@@ -0,0 +1,204 @@
+
+## Deployment Topology
+
+This graphic shows the CI/CD workflow topology needed by our enterprise customers to deploy infrastructure to Azure. The journey starts with a git commit to the IAC repo. This workflow validates and deploys the desired infrastructure state across the azure environments.
+
+![infrastructure_reference_cicd_flow](https://user-images.githubusercontent.com/7635865/60440330-3a039580-9bda-11e9-8af6-a883739c2d87.jpg)
+
+## In Scope
+
+- Azure DevOps build template
+- Sequence diagram showing the journey with sourcing, testing, building and releasing IAC(Infra-as-code) templates across azure environment stages.
+
+## Out of Scope
+
+- Observability
+- Monitoring
+- Azure Boards
+- Automating the provisioning of the azure devops pipeline prerequisites
+- Automated load testing
+- Chaos testing
+- Active Directory Setup
+- Multi-region deployments
+- Build Agent Pool Setup
+- Remote terraform modules for auto scaling and monitoring rules
+
+## Customers
+
+- **Technical Operators**: This template uses a service principal to carry out the terraform resource provisioning. The deployment service principal will need contributor role level permissions on the cloud resources.
+
+    Provisioning templates that modify role assignments will require owner level access to the target resource(s).
+
+## Security
+
+Deployments are carried out by service principals. The build pipeline references SP credentials from a keyvault-backed variable group.
+
+You'll need to create a keyvault resource that includes the service principal credential secrets listed below.
+
+### Required Key Vault Secrets
+
+- `AD-SP-CLIENT-ID` - The Azure service principal client id used for the deployment.
+- `AD-SP-SECRET` - The Azure service principal secret used for the deployment.
+- `AD-SP-SUBSCRIPTION-ID` - The Azure subscription of the service principal used for the deployment.
+- `AD-SP-TENANT-ID` - The Azure service principal tenant id used for the deployment.
+- `ARM-ACCESS-KEY` - The remote state storage account access key used for the deployment.
+
+Follow these [instructions](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/variable-groups?view=azure-devops&tabs=yaml#link-secrets-from-an-azure-key-vault) to associate all the above secrets to a variable group called `KV Secrets`.
+
+## Prerequisites
+
+- An provisioned Azure DevOps instance
+- Service principal
+- Storage account for tracking remote terraform state. We recommend [backing](https://azure.microsoft.com/en-us/services/backup/) this storage account up in Azure. Terraform will delete all provisioned resources if the remote state file is either corrupted or deleted.
+- Key Vault resource including the service principal and storage account credentials following these [instructions](../../README.md#required-key-vault-secrets).
```

Comment:
It would be awesome to summarize the variables required by the pipeline along with the KV secrets in a table with the following columns:

Variable Name, Variable Source (KV or Pipeline), sourced from location, Description/Use. I think it would flow most naturally after the description of the build/CD pipelines.